lib-utilities/services/rpcService.go

```diff
@@ -15,66 +15,265 @@
 package services
 
 import (
+	"context"
 	"crypto/tls"
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
+	"net"
+	"time"
 
 	"github.com/ODIM-Project/ODIM/lib-utilities/config"
+	"github.com/coreos/etcd/clientv3"
 	micro "github.com/micro/go-micro"
 	"github.com/micro/go-micro/transport"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
+// frameworkType is for selecting framework for bringing up the microservices
+type frameworkType int
+
+const (
+	// GRPC allows the microservices to be brought up with gRPC protocol
+	GRPC frameworkType = iota
+	// GoMicro allows the microservices to be brought up with GoMicro framework
+	GoMicro
+)
+
+type serviceType int
+
+const (
+	serverService serviceType = iota
+	clientService
+)
+
+// odimService holds the components for bringing up and communicating with a micro service
+type odimService struct {
+	clientTransportCreds credentials.TransportCredentials
+	registryAddress      string
+	Server               *grpc.Server
+	serverAddress        string
+	serverName           string
+	serverTransportCreds credentials.TransportCredentials
+}
+
+// ODIMService holds the initialized instance of odimService
+var ODIMService odimService
+
 // Service holds the microservice instance
 var Service micro.Service
 
 // InitializeService will initialize a new micro.Service.
 // Service will be initialized here itself, so the Server() and Client()
 // called easily.
-func InitializeService(serviceName string) error {
+func InitializeService(framework frameworkType, serverName string) error {
+	switch framework {
+	case GRPC:
```

Comment:
This is missing TLS config

Reply:
@Bharath-KKB TLS implementation is moved to init() function.